### PR TITLE
Issue 626 by pjohans: update ting_object table to allow multiple NULL va...

### PR DESCRIPTION
**Pull request by pjohans**
...lues in vid-field.

Basically this is a copy of https://api.drupal.org/api/drupal/modules!node!node.install/function/node_update_7013/7 as suggested by Kasper https://github.com/ding2/ting/pull/13

### CODE REVIEW

File: ting.install [hunk 1]
```diff
@@ -338,3 +338,19 @@ function ting_update_7008() {
...
+  db_drop_unique_key('ting_object', 'vid');
+  db_change_field('ting_object', 'vid', 'vid', array(
+    'description' => 'The current {ting_obuect}.vid version identifier.',
+    'type' => 'int',
+    'unsigned' => TRUE,
+    'not null' => FALSE,
+    'default' => NULL,
+  ));
+  db_add_unique_key('ting_object', 'vid', array('vid'));
+}
```
Comment (kasperg) [inline ting.install]: There is a missing newline at the end of the file.

File: ting.install [hunk 2]
```diff
@@ -338,3 +338,19 @@ function ting_update_7008() {
...
 }
+
+/**
+ * Update ting_object table - set foreign key (vid) to ting_object_revision to be null by default thus allowing
+ * multiple NULL values in table
+ */
+function ting_update_7009() {
+  db_drop_unique_key('ting_object', 'vid');
+  db_change_field('ting_object', 'vid', 'vid', array(
+    'description' => 'The current {ting_obuect}.vid version identifier.',
```
Comment (kasperg) [inline ting.install]: Minor issue: There is a typo in `ting_obuect`.

**Comment by kasperg** [comment]
This needs to be changed in [`ting_schema()`](https://github.com/ding2/ting/blob/master/ting.install#L27-28) to take effect on new installations.

**Comment by kasperg** [comment]
Review complete.

To merge this we at least have to update `ting_schema()`. While at it the other two minor issues would be good to fix the two minor issues as well.

**Comment by pjohans** [comment]
i'm on it

**Comment by pjohans** [comment]
fixed

**Comment by kasperg** [comment]
Great. Approved and merged.

**Pull request merged**